doc/source/development/contributing_environment.rst

```diff
@@ -164,7 +168,11 @@ You also need to have ``setuptools`` 51.0.0 or later to build pandas.
 
    # Build and install pandas
    python setup.py build_ext -j 4
-   python -m pip install -e . --no-build-isolation --no-use-pep517
+   python setup.py develop
+
+   # Alternatively, if you would like to try out our new build system
+   # based on meson
+   pip install . --no-build-isolation --config-settings="builddir=builddir"
```

Comment:
Is the idea that we can just run this by itself? Tried and when importing pandas after build it errored as if the extensions were not built

Reply:
You can't import from the base of the source directory. If you do, python will automatically add the current directory to the front of PYTHONPATH (I don't think there's any way to remove it), and it'll try to import the missing extensions(since they're not built inplace).

I don't think there's a really good solution for this right now. All the solutions I can think of right now involve editing __init__.py, and that's tricky(since we'll need to have the edited version in the source directory that imports them from site-packages, but the original version in the installed pandas).

The editable installs meson-python PR is moving along again right now, so I'd rather just wait for that, and see what more needs to be done from there.

Reply:
Ah OK. I guess this makes sense since there is no inplace build with meson